pandas/io/formats/format.py

```diff
@@ -1622,17 +1622,30 @@ def __init__(
     def _format_strings(self) -> list[str]:
         """we by definition have DO NOT have a TZ"""
         values = self.values
+        values_shape = values.shape
+        values = values.ravel() if values.ndim > 1 else values
 
         if not isinstance(values, DatetimeIndex):
             values = DatetimeIndex(values)
 
         if self.formatter is not None and callable(self.formatter):
-            return [self.formatter(x) for x in values]
+            values = [self.formatter(x) for x in values]
+        else:
+            values = values._data._format_native_types(
+                na_rep=self.nat_rep, date_format=self.date_format
+            )
 
-        fmt_values = values._data._format_native_types(
-            na_rep=self.nat_rep, date_format=self.date_format
-        )
-        return fmt_values.tolist()
+        if len(values_shape) > 1:
+            fmt_values = np.reshape(values, values_shape)
+            nested_formatter = GenericArrayFormatter(fmt_values)
+            fmt_values = nested_formatter.get_result()
+        else:
+            fmt_values = values
+
+        if isinstance(fmt_values, np.ndarray):
```

Comment:
what else can these be? can we just `list(fmt_values)`?

Reply:
Hi Yes. It can. Have updated the code.